crates/typst-library/src/compute/calc.rs

```diff
@@ -188,6 +189,34 @@ pub fn sqrt(
     Ok(value.v.float().sqrt())
 }
 
+/// Calculates the nth root of a number.
+///
+/// ```example
+/// #calc.root(4, 16) \
+/// #calc.root(3, 27)
+/// ```
+#[func]
+pub fn root(
+    /// Which root of the randicand to take
+    index: Spanned<i64>,
+    /// The expression to take the root of
+    radicand: Spanned<Num>,
+) -> SourceResult<f64> {
+    if index.v == 0 {
+        bail!(index.span, "cannot take the 0th root of a number");
+    } else if radicand.v.float() < 0. {
+        if index.v % 2 == 0 {
+            bail!(index.span, "cannot take the even root of a negative number because the result it is not real");
```

Comment:
It leads to think that it is a complex number, which is not the case. For instance sqrt of `-1` is not equal to `i` (you can prove this by contradiction), for even roots it is only defined for positive numbers.

Reply:
I'm not quite certain what you mean. What do you mean the square root of -1 is not i? Is that not the definition of imaginary numbers?

Reply:
The definition of `i` is `i^2=1`. It is not the same thing as `i=sqrt(-1)`. (basically because sqrt is not defined for negative numbers, but it would also make incoherences, when I got some time I'll write a basic example if you want)

Reply:
Here is a short proof : ![rendered.png](https://github.com/typst/typst/assets/23212967/e6a20c63-bfcb-4d67-ad29-67ccbd8f0659)

Reply:
Well, from what I understand the fallacy is that $\sqrt{-1} \cdot \sqrt{-1} = \sqrt{-1 \cdot -1}$ because the product property of square roots, $\sqrt{ab} = \sqrt{a} \sqrt{b}$ only holds for $a, b \geq 0$

Reply:
Alriht, I think I finally understand what you're trying to convey, correct me if I'm wrong. You say `sqrt(-1)` is undefined, so saying that the even root of a negative number is not real implies that it is complex, which you don't agree with.

Reply:
Mathematically it’s perfectly fine for `sqrt(-1)` to be defined as `i`, it’s called a [principal value](https://en.wikipedia.org/wiki/Principal_value).

Reply:
> Here is a short proof : ![rendered.png](https://github.com/typst/typst/assets/23212967/e6a20c63-bfcb-4d67-ad29-67ccbd8f0659)

i is the principal value of sqrt(-1) (the other is -i)

Your proof is incorrect because the identity sqrt(ab)=sqrt(a)sqrt(b) does not hold for complex numbers with arbitrary argument